MNT modify test_nested_cv to speed-up CI builds

Shuyib opened the pull request:
#### Reference Issues/PRs
References #21407

#### What does this implement/fix? Explain your changes.
Edit the LeaveOneOut() model validation strategy, reduce the max_iter=1 for Ridge regression and adjust nsplits=2 for other evaluation strategies to speed up runtime to under 1s.


#### Any other comments?
Pair programming partner @Lontia
#DataUmbrella Sprint

Shuyib committed:
Update test_split.py

Remove the LeaveOneOut() cross-validation strategy, adjust the n_splits=2 and adjust the max_iter=1 in Ridge regression.

glemaitre commented:
The linter is complaining. Could you make sure to run black on the file that you modified.

glemaitre commented on sklearn/model_selection/tests/test_split.py:
```diff
@@ -1771,16 +1771,15 @@ def test_nested_cv():
     for inner_cv, outer_cv in combinations_with_replacement(cvs, 2):
         gs = GridSearchCV(
-            Ridge(), param_grid={"alpha": [1, 0.1]}, cv=inner_cv, error_score="raise"
+            Ridge(max_iter=1), param_grid={"alpha": [1, 0.1]}, cv=inner_cv, error_score="raise"
```
I would not change the number of iteration here. It will generate some new `ConvergenceWarning`. Reducing the number of splits seem the best.

glemaitre replied:
Another possibility would be to use a `DummyClassifier` and tune two simple strategies. It will be extremely fast and will be enough for this smoke test.

Shuyib replied:
Hi @glemaitre I see there that I had called the Ridge() twice and I realized that this will definitely fail. I did this late last night therefore I seem to missed that. Apologies editing the file now.

Shuyib committed:
Update test_split.py

- Update the model_validation strategies to only 2.
- Replace the classifier from Ridge to DummyClassifier.
- Edit the parameter grid for the classifier. Model runs for 0.07s on average using 3 using three tests.

Shuyib committed:
Merge branch 'scikit-learn:main' into test_nested_csv_edit

Shuyib committed:
Update test_split.py

edit the gridSearchCV definition in test_nested_cv.

Shuyib committed:
Update test_split.py

remove whitespace.

thomasjpfan reviewed (commented):
There are [linting errors](https://dev.azure.com/scikit-learn/scikit-learn/_build/results?buildId=34400&view=logs&j=32e2e1bb-a28f-5b18-6cfc-3f01273f5609&t=fc67071d-c3d4-58b8-d38e-cafc0d3c731a). You can resolve them by installing black and running it on the codebase:

```bash
pip install black==21.6b0
black .
```

ogrisel commented:
@Shuyib it was not necessary to create a new PR, you could have pushed to the same branch on your fork to automatically have github update this PR with your latest changes but this is fine. Let me close this PR in favor of #21551.
